### ISSUE TITLE
InVEST 3.9 Windows installer NDR and SWY crash on sample data

### ISSUE THREAD

**Issue by dcdenu4** [natcap/invest#321]
While testing / debugging PR #314 we noticed that NDR and SWY were failing on sample data. After testing out prior versions of the 3.9 builds it was shown that these errors have been hiding there for at least a little while. So far it is unclear when these models started failing on the sample data as the testing framework did not catch them.

NDR:
```python
Traceback (most recent call last):
  File "site-packages\natcap\invest\ui\model.py", line 1629, in _logged_target
  File "site-packages\natcap\invest\ndr\ndr.py", line 490, in execute
  File "site-packages\taskgraph\Task.py", line 697, in add_task
  File "site-packages\taskgraph\Task.py", line 1174, in _call
  File "site-packages\natcap\invest\ndr\ndr.py", line 900, in _normalize_raster
  File "site-packages\pygeoprocessing\geoprocessing.py", line 433, in raster_calculator
  File "site-packages\natcap\invest\ndr\ndr.py", line 887, in _normalize_raster_op
IndexError: boolean index did not match indexed array along dimension 0; dimension is 256 but corresponding boolean dimension is 91
```

SWY:
```python
Traceback (most recent call last):
  File "site-packages\natcap\invest\ui\model.py", line 1648, in _logged_target
  File "site-packages\natcap\invest\seasonal_water_yield\seasonal_water_yield.py", line 377, in execute
  File "site-packages\natcap\invest\seasonal_water_yield\seasonal_water_yield.py", line 644, in _execute
  File "site-packages\taskgraph\Task.py", line 697, in add_task
  File "site-packages\taskgraph\Task.py", line 1174, in _call
  File "site-packages\natcap\invest\seasonal_water_yield\seasonal_water_yield.py", line 1073, in _calculate_curve_number_raster
  File "site-packages\pygeoprocessing\geoprocessing.py", line 433, in raster_calculator
  File "site-packages\natcap\invest\seasonal_water_yield\seasonal_water_yield.py", line 1055, in cn_op
ValueError: Values in the LULC raster were found that are not represented under the 'lucode' key column of the Biophysical table. The missing values found in the LULC raster but not the table are: [-999999].
```

The 3.9 build I used was https://github.com/natcap/invest/actions/runs/268620690

**Comment by dcdenu4** [comment]
I think these are the changes that cause the sample data to fail with NDR:https://github.com/natcap/invest/commit/e3f95152f9dd1b5fcae711f4106f9623a64115df

**Comment by phargogh** [comment]
Well, shoot.  This is a good legitimation for running the models on the sample data as part of our regular testing.

**Comment by dcdenu4** [comment]
The NDR error looks like a typo in this chunk of code where `raster_block` should be `array`:
```python
    def _normalize_raster_op(array):
        """Divide values by mean."""
        result = numpy.empty(array.shape, dtype=numpy.float32)
        result[:] = numpy.float32(base_nodata)

        valid_mask = slice(None)
        if base_nodata is not None:
            #valid_mask = ~numpy.isclose(raster_block, base_nodata)
            valid_mask = ~numpy.isclose(array, base_nodata)
        result[valid_mask] = array[valid_mask]
        if value_mean != 0:
            result[valid_mask] /= value_mean
        return result
```

**Comment by dcdenu4** [comment]
There was a bug in SWY where when reclassifying, the nodata value of the LULC was not being taken into account when comparing to the value map.

**Comment by dcdenu4** [comment]
Fixed in #323
